src/BuiltInTools/dotnet-watch/Internal/PhysicalConsole.cs

```diff
@@ -43,10 +43,13 @@ private void ListenToConsoleKeyPress()
             {
                 while (true)
                 {
-                    var key = Console.ReadKey(intercept: true);
-                    for (var i = 0; i < _keyPressedListeners.Count; i++)
+                    if (Console.KeyAvailable)
```

Comment:
>Console [](http://example.com/codeflow?start=24&length=7)

This doesn't look right. Are we gonna be spinning the CPU at 100% until there is a key available?

Reply:
Yeah, you are right. I missed that. 
I've tried to play around with the Process.StandartInput stream but it causes more problems.
Spinning CPU is bad, but I don't see another solution

Reply:
Of course, we can call something like `Thread.Sleep(100)` but this will not look good too